polynote-kernel/src/main/scala/polynote/kernel/LocalKernel.scala

```diff
@@ -49,25 +49,23 @@ class LocalKernel private[kernel] (
             _             <- busyState.update(_.setBusy)
             notebook      <- CurrentNotebook.get
             cell          <- ZIO(notebook.cell(id))
-            interpEnv     <- InterpreterEnvironment.fromKernel(id)
-            interpreter   <- getOrLaunch(cell.language, CellID(0)).provideSomeM(Env.enrich[BaseEnv with GlobalEnv with CellEnv](interpEnv: InterpreterEnv))
+            interpreter   <- getOrLaunch(cell.language, CellID(0))
             state         <- interpreterState.get
             prevCells      = notebook.cells.takeWhile(_.id != cell.id)                                                  // find the latest executed state that correlates to a notebook cell
             prevState      = prevCells.reverse.map(_.id).flatMap(state.at).headOption.getOrElse(latestPredef(state))
             _             <- PublishResult(ClearResults())
             _             <- PublishResult(ExecutionInfo(startTime, None))
             _             <- CurrentNotebook.get
             initialState   = State.id(id, prevState)                                                                    // run the cell while capturing outputs
-            resultState   <- (interpreter.run(cell.content.toString, initialState) >>= updateValues)
+            resultState   <- (interpreter.run(cell.content.toString, initialState).provideSomeLayer(CellExecutor.layer(scalaCompiler.classLoader)) >>= updateValues)
               .ensuring(CurrentRuntime.access.flatMap(rt => ZIO.effectTotal(rt.clearExecutionStatus())))
-              .provideSomeM(Env.enrichM[Logging with Clock](interpEnv.mkExecutor(scalaCompiler.classLoader).widen[InterpreterEnv]))
             _             <- publishEndTime
             _             <- updateState(resultState)
             _             <- resultState.values.map(PublishResult.apply).sequence.unit                                  // publish the result values
             _             <- busyState.update(_.setIdle)
           } yield ()
 
-          run.supervised.onInterrupt {
+          run.provideSomeLayer(CurrentRuntime.layer(id)).onInterrupt { _ =>
```

Comment:
Just curious, why did you move this environment enrichment here rather than up where resultState is defined?